internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_availability_set_resource_test.go

```diff
@@ -0,0 +1,210 @@
+type SystemCenterVirtualMachineManagerAvailabilitySetResource struct{}
+
+func TestAccSystemCenterVirtualMachineManagerAvailabilitySetSequential(t *testing.T) {
+	// NOTE: this is a combined test rather than separate split out tests because only one System Center Virtual Machine Manager Server can be onboarded at a time on a given Custom Location
+
+	if os.Getenv("ARM_TEST_CUSTOM_LOCATION_ID") == "" || os.Getenv("ARM_TEST_FQDN") == "" || os.Getenv("ARM_TEST_USERNAME") == "" || os.Getenv("ARM_TEST_PASSWORD") == "" {
+		t.Skip("Skipping as one of `ARM_TEST_CUSTOM_LOCATION_ID`, `ARM_TEST_FQDN`, `ARM_TEST_USERNAME`, `ARM_TEST_PASSWORD` was not specified")
+	}
+
+	acceptance.RunTestsInSequence(t, map[string]map[string]func(t *testing.T){
+		"scvmmAvailabilitySet": {
+			"basic":          testAccSystemCenterVirtualMachineManagerAvailabilitySet_basic,
+			"requiresImport": testAccSystemCenterVirtualMachineManagerAvailabilitySet_requiresImport,
+			"complete":       testAccSystemCenterVirtualMachineManagerAvailabilitySet_complete,
+			"update":         testAccSystemCenterVirtualMachineManagerAvailabilitySet_update,
+		},
+	})
+}
+
+func testAccSystemCenterVirtualMachineManagerAvailabilitySet_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_system_center_virtual_machine_manager_availability_set", "test")
+	r := SystemCenterVirtualMachineManagerAvailabilitySetResource{}
+
+	data.ResourceSequentialTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func testAccSystemCenterVirtualMachineManagerAvailabilitySet_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_system_center_virtual_machine_manager_availability_set", "test")
+	r := SystemCenterVirtualMachineManagerAvailabilitySetResource{}
+
+	data.ResourceSequentialTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func testAccSystemCenterVirtualMachineManagerAvailabilitySet_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_system_center_virtual_machine_manager_availability_set", "test")
+	r := SystemCenterVirtualMachineManagerAvailabilitySetResource{}
+
+	data.ResourceSequentialTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func testAccSystemCenterVirtualMachineManagerAvailabilitySet_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_system_center_virtual_machine_manager_availability_set", "test")
+	r := SystemCenterVirtualMachineManagerAvailabilitySetResource{}
+
+	data.ResourceSequentialTest(t, r, []acceptance.TestStep{
```

Comment:
To test `add`, `update` and `removal` of updatable properties, shall we test with `basic -> complete -> update -> basic`?

Reply:
Updated